rotors_gazebo_plugins/src/gazebo_wind_plugin.cpp

```diff
@@ -145,5 +236,120 @@ void GazeboWindPlugin::OnUpdate(const common::UpdateInfo& _info) {
+  } else {
+    gzerr << "Could not open custom wind field text file.\n";
+  }
+
+}
+
+/*
+  Linear interpolation
+    position: y-coordinate of the target point
+    values:   a pointer to an array of size 2 containing the values of the two points to interpolate from (12 and 13)
+    points:   a pointer to an array of size 2 containing the y-coordinate of the two points to interpolate from
+*/
+math::Vector3 GazeboWindPlugin::LinearInterpolation(double position, math::Vector3 *values, float *points) {
+  math::Vector3 value = values[0] + (values[1] - values[0]) / (points[1] - points[0]) * (position - points[0]);
+  return value;
+}
+
+/*
+  Bilinear interpolation
+    position: a pointer to an aray of size 2 containing the x and y-coordinates of the target point
+    values:   a pointer to an array of size 4 containing the values of the four points to interpolate from (8, 9, 10 and 11)
+    points:   a pointer to an array of size 6 containing the x-coordinate of the four points to interpolate from,
+              and the y-coordinate of the two intermediate points (12 and 13)
+*/
+math::Vector3 GazeboWindPlugin::BilinearInterpolation(double *position, math::Vector3 *values, float *points) {
+  math::Vector3 intermediate_values[2] = {
+                                            LinearInterpolation(position[0],&(values[0]),&(points[0])),
+                                            LinearInterpolation(position[0],&(values[2]),&(points[2]))
+                                         };
+  math::Vector3 value = LinearInterpolation(position[1], intermediate_values, &(points[4]));
+  return value;
+}
+
+/*
+  Trilinear interpolation
+    link_position:  a Vector3 containing the x, y and z-coordinates of the target point
+    values:         a pointer to an array of size 8 containing the values of the eight points to interpolate from (0, 1, 2, 3, 4, 5, 6 and 7)
+    points:         a pointer to an array of size 14 containing the z-coordinate of the eight points to interpolate from,
+                    the x-coordinate of the four intermediate points (8, 9, 10 and 11), and the y-coordinate of the last two intermediate points (12 and 13)
+*/
+math::Vector3 GazeboWindPlugin::TrilinearInterpolation(math::Vector3 link_position, math::Vector3 *values, float *points) {
+  double position[3] = {link_position.x,link_position.y,link_position.z};
+  math::Vector3 intermediate_values[4] = {
+                                            LinearInterpolation(position[2],&(values[0]),&(points[0])),
+                                            LinearInterpolation(position[2],&(values[2]),&(points[2])),
+                                            LinearInterpolation(position[2],&(values[4]),&(points[4])),
+                                            LinearInterpolation(position[2],&(values[6]),&(points[6]))
+                                         };
+  math::Vector3 value = BilinearInterpolation(&(position[0]),intermediate_values,&(points[8]));
```

Comment:
I see what you mean, yes, I have thought about it. It would save me 6 lines of code and 3 of comments, and cost me no line of code and only one or two lines of comments.

However, I think I prefer sticking with the current version and keeping a more uniform and consistent code.